### PR TITLE
[MXNET-1405] tests for large tensor support for Softmax operator

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
Added test case to verify Large Tensor Support for Softmax Operator

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [MXNET-1405], where #15042 refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by access2rohit** [comment]
@apeforest : Please review

File: tests/nightly/test_large_array.py [hunk 1]
```diff
@@ -279,6 +279,19 @@ def test_diag():
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
 
+def test_softmax():
+    def softmax_forward(input_data, true_output):
+        data = mx.sym.Variable('data')
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: Why not just use ndarray API?

File: tests/nightly/test_large_array.py [hunk 2]
```diff
@@ -279,6 +279,19 @@ def test_diag():
...
+        out1 = data.softmax(axis=0)
+        exec1 = out1.bind(mx.cpu(), args={'data': input_data})
+        exec1.forward()[0].wait_to_read()
+        ndarr = exec1.outputs[0]
+        nparr = ndarr.asnumpy()
+        assert_almost_equal(nparr, true_output, rtol=1e-5, atol=1e-5)
+
+    softmax_forward(mx.nd.ones((128, LARGE_X)), np.full((128, LARGE_X), 0.0078125))
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: Is this 0.0078125 hand calculated? How to verify it? Can you use numpy calculation as the expected result?

Comment (apeforest) [inline tests/nightly/test_large_array.py]: Why 128? Can we use predefined constants?

File: tests/nightly/test_large_array.py [hunk 3]
```diff
@@ -279,6 +279,19 @@ def test_diag():
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
 
+def test_softmax():
+    def softmax_forward(input_data, true_output):
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: If this is called once, do we need this sub routine?

File: tests/nightly/test_large_array.py (hunk 2 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: Yes ... will do

File: tests/nightly/test_large_array.py (hunk 3 above)
Comment (access2rohit) [reply tests/nightly/test_large_array.py]: ok

File: tests/nightly/test_large_array.py [hunk 4]
```diff
@@ -279,6 +272,13 @@ def test_diag():
     assert_almost_equal(r.asnumpy(), np.diag(a_np, k=k))
 
 
+def test_softmax():
+    input_data = mx.nd.ones((SMALL_Y, LARGE_X))
+    true_output = np.full((SMALL_Y, LARGE_X), (1 / SMALL_Y))
+    output = nd.softmax(input_data, axis=0)
```
Comment (apeforest) [inline tests/nightly/test_large_array.py]: when you test this way, the softmax is operating only on one dimension which does not exceed 2^32, right?

Comment (access2rohit) [reply tests/nightly/test_large_array.py]: yes

Comment (apeforest) [reply tests/nightly/test_large_array.py]: But shouldn't we test softmax over a vector larger than 2^32?

Comment (access2rohit) [reply tests/nightly/test_large_array.py]: The test fails

Comment (apeforest) [reply tests/nightly/test_large_array.py]: fine. let's leave that as TODO in the next phase to support large int in one dimension.

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-merge]

**Pull request merged**